docs/blog/2025-03-19-using-angular-with-rspack.md

```diff
@@ -0,0 +1,166 @@
+cover_image: /blog/images/2025-03-19/angular-rspack-logo-small.png
+description: Learn about how and why to use Rspack with Angular thanks to Nx's efforts for supporting Rspack for Angular.
+---
+
+Building applications with [Angular](https://angular.dev) has always been a lesser concern for most Angular developers due to the abstractions that Angular created called `builders`. The underlying implementation details were hidden from the developer who only needed to run either `ng build` or `nx build`.  
+Despite this, most Angular developers knew that it was originally [webpack](https://webpack.js.org/) that was used to build their applications. This was a great solution at the time and it was even possible to extend their builds by leveraging custom webpack configurations and plugins.
+
+Over time, as applications grew in size and complexity, it became clear that the inherit slowness with Webpack build speeds was becoming more and more of an issue for Angular developers.
+
+The Angular Team decided to address this build speed issue by building out a new build pipeline that leveraged [Esbuild](https://esbuild.github.io/).
+
+This succeeded in reducing the build times for Angular applications, however, it made one crucial mistake. It left the existing Angular applications that relied on the Webpack ecosystem behind, with either a difficult migration path or none at all.
+To this day, many existing Angular applications still rely on Webpack because they cannot readily replace their Webpack configurations and plugins with equivalent Esbuild plugins. Thus, they are continuing to fight with slow builds reducing their productivity.
+
+## What about Rspack?
+
+[Rspack](https://rspack.dev) is a high performance JavaScript bundler written in Rust. It offers strong compatibility with the Webpack ecosystem, allowing for almost seamless replacement of webpack, and provides lightning fast build speeds.
+
+Because it supports the existing Webpack ecosystem, it provides an answer to teams that maintain Angular applications using Webpack and want to migrate to a faster build pipeline.
+
+However, it is crucial to understand that Rspack is not _completely_ compatible with the Webpack ecosystem with some slight nuances and low-level api differences that prevent certain plugins and loaders from working out of the box.
+One such example is the `AngularIvyPlugin` which is a Webpack plugin that is used to support Angular's Ivy compiler. Therefore, it was not possible to simply drop in Rpsack and expect it to work with Angular applications.
+Many people have tried to get Rspack working with Angular, but it has proven to be a challenge, with partial support and success being reported in the past but with a lot of limitations or actual performance degradations over the Webpack approach.
+
+Instead, a new approach was needed to support Rspack with Angular. A closer examination and replication of how Angular compiles and bundles the application was required.
```

Comment:
This sentence is kinda weird. I get what it is hinting at, but can we be more specific. Like saying current tentatives focused on ..... while instead it was necessary to .... (which Angular Rspack does)

Reply:
I will have a think about how to rephrase it